### PR TITLE
adds EncryptionConfig To GroupResourceKeys roundtrip tests

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/encryption/state_test.go [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+					},
+				},
+
+				{Group: "", Resource: "configmaps"}: {
+					writeKey: keyAndMode{
+						key: apiserverconfigv1.Key{Name: "", Secret: ""}, mode: "identity",
+					},
+					readKeys: []keyAndMode{
+						{key: apiserverconfigv1.Key{Name: "34", Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc="}, mode: "aescbc"},
+						{key: apiserverconfigv1.Key{Name: "35", Secret: newFakeIdentityEncodedKeyForTest()}, mode: "aescbc"},
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/state_test.go]: this should be `aesgcm` - I need to fix that.

Comment (sttts) [reply pkg/operator/encryption/state_test.go]: Is this a `/hold`?

Comment (p0lyn0mial) [reply pkg/operator/encryption/state_test.go]: ok, let's put it into `hold`, but please feel free to review it anyway.

File: pkg/operator/encryption/state_test.go [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+			input: func() *apiserverconfigv1.EncryptionConfiguration {
+				keysRes := encryptionKeysResourceTuple{
+					resource: "secrets",
+					keys: []apiserverconfigv1.Key{
+						{
+							Name:   "34",
+							Secret: "MTcxNTgyYTBmY2Q2YzVmZGI2NWNiZjVhM2U5MjQ5ZDc=",
+						},
+
+						// secretsToProviders puts "fakeIdentityProvider" as last
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/state_test.go]: will this special key be ever removed?
what happens when a user frequently turns on and off the encryption?

Comment (p0lyn0mial) [reply pkg/operator/encryption/state_test.go]: would be nice to add unit tests for ^

File: pkg/operator/encryption/helpers_test.go [hunk 3]
```diff
@@ -238,9 +281,13 @@ func createEncryptionCfgSecretWithWriteKeys(t *testing.T, targetNs string, revis
 	}
 }
 
+// TODO: change the name to encryptionKeysResourceMode
 type encryptionKeysResourceTuple struct {
```
Comment (sttts) [inline pkg/operator/encryption/helpers_test.go]: we have too many key/mode structs

File: pkg/operator/encryption/state_test.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/encryption/state_test.go]: I don't understand that well yet. Yes, at the very least we need heavy unit tests.

**Pull request closed without merging**